File: src/main/java/com/example/temperatura/converter/interfaces/impl/CalculadoraTemperaturaImpl.java
package com.example.temperatura.converter.interfaces.impl;

import com.example.temperatura.converter.interfaces.CalculadoraTemperatura;

public class CalculadoraTemperaturaImpl implements CalculadoraTemperatura {

	public Double celsiusToFarenheit(Double tempCelsius) {
		return (tempCelsius * 9/5) + 32;
	}

	public Double celsiusToKelvin(Double tempCelsius) {
		return tempCelsius + 273.15;
	}

	public Double farenheitToCelsius(Double tempFarenheit) {
		return ((tempFarenheit - 32) * 5)/9;
	}

	public Double farenheitToKelvin(Double tempFarenkeit) {
		return ((tempFarenkeit - 32) * 5)/9 + 273.15;
	}

	public Double kelvinToCelsius(Double tempKelvin) {
		return tempKelvin - 273.15;
	}

	public Double kelvinToFarenheit(Double tempKelvin) {
		return ((tempKelvin - 273.15) * 9)/5 + 32;
	}

}

File: src/main/kotlin/com/example/temperatura/converter/interfaces/impl/CalculadoraTemperaturaImpl.kt
package com.example.temperatura.converter.interfaces.impl

import com.example.temperatura.converter.interfaces.CalculadoraTemperatura
import org.springframework.stereotype.Service

@Service
class CalculadoraTemperaturaImpl : CalculadoraTemperatura {
    override fun celsiusToFarenheit(tempCelsius: Double): Double = (tempCelsius * 9 / 5) + 32

    override fun celsiusToKelvin(tempCelsius: Double): Double = tempCelsius + 273.15

    override fun farenheitToCelsius(tempFarenheit: Double): Double = ((tempFarenheit - 32) * 5) / 9

    override fun farenheitToKelvin(tempFarenheit: Double): Double = (tempFarenheit - 32) * 5 / 9 + 273.15

    override fun kelvinToCelsius(tempKelvin: Double): Double = tempKelvin - 273.15

    override fun kelvinToFarenheit(tempKelvin: Double): Double = (tempKelvin - 273.15) * 9 / 5 + 32
}